### PR TITLE
ZIOS-11281: Delivery receipt - send a cumulative one once the sync is done

### PR DESCRIPTION
**Pull request by nicorsm**
## What's new in this PR?

- Added sending of a single delivery receipt, like it's already happening with reading receipts (it uses the same method)
- Added test
- Fixed a repetition in a method signature

### CODE REVIEW

**Comment by vytis** [comment]
I guess it's worth making a playground build with this change to check if other platforms support batching of delivery receipts also.

File: Source/Model/Conversation/ZMConversation+Timestamps.swift [hunk 1]
```diff
@@ -277,6 +277,7 @@ extension ZMConversation {
     func savePendingLastRead() {
         guard let timestamp = pendingLastReadServerTimestamp else { return }
         confirmUnreadMessagesAsRead(until: timestamp)
+        confirmMessagesAsDelivered()
```
Comment (typfel) [inline Source/Model/Conversation/ZMConversation+Timestamps.swift]: There's a difference between `read` and `delivery` receipts: delivery receipts should be sent in the background without any action of the receiver. `savePendingLastRead()` is called as a response to opening and reading a conversation.

File: Source/Model/Conversation/ZMConversation+Confirmations.swift [hunk 2]
```diff
@@ -44,6 +44,34 @@ extension ZMConversation {
         return confirmationMessages
     }
     
+    public static func confirmDeliveredMessages(_ messages: [UUID], in conversations: [UUID], with managedObjectContext: NSManagedObjectContext) -> [ZMMessage] {
+        
+        var confirmationMessages: [ZMMessage] = []
+        for conversationID in conversations {
+            guard let convo = ZMConversation(remoteID: conversationID, createIfNeeded: false, in: managedObjectContext),
```
Comment (typfel) [inline Source/Model/Conversation/ZMConversation+Confirmations.swift]: We could use `ZMManagedObject.fetchObjectsWithRemoteIdentifiers()` to fetch all conversations with one fetch request. Look at how it's done for users in `ZMuser.usersWithRemoteIDs()`

We could the same thing for messages.

**Comment by nicorsm** [comment]
@zenkins release minor

**Pull request merged**